Shopify/cli
# Set cloudflare as default & add tunnel flags

### WHY are these changes introduced?
Contributes to https://github.com/Shopify/internal-cli-foundations/issues/307



### WHAT is this pull request doing?
- Set cloudflare as default tunnel solution
- Add a new (hidden) flag to select between ngrok and cloudflare
- Select the tunnel provider based on a remote feature flag


### How to test your changes?
- `app dev` should automatically create a cloudflare tunnel
- `app dev --tunnel ngrok` should create an ngrok tunnel


### Post-release steps
Update documentation saying:
- We use cloudlflare by default, there is no need for auth or anything, it'd just work
- ngrok is optional. Use it with `--tunnel ngrok`



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
